src/nomad_catalysis/schema_packages/catalysis.py

```diff
@@ -0,0 +1,378 @@
+    )
+
+    def populate_results(
+            self, archive: 'EntryArchive', logger) -> None:
+        '''
+        This function copies the catalyst sample information specified in the dict
+         quantities_results_mapping into the results archive of the entry.
+        '''
+
+        sample_obj = self
+
+        add_catalyst(archive)
+        quantities_results_mapping = {
+            'name': 'catalyst_name',
+            'catalyst_type': 'catalyst_type',
+            'preparation_details.preparation_method': 'preparation_method',
+            'surface.surface_area': 'surface_area',
+            'surface.method_surface_area_determination': 'characterization_methods',
+        }
+
+        for ref_attr, catalyst_attr in quantities_results_mapping.items():
+            value = get_nested_attr(sample_obj, ref_attr)
+            if value is not None:
+                try:
+                    setattr(
+                        archive.results.properties.catalytic.catalyst,
+                        catalyst_attr,
+                        value
+                    )
+                except ValueError:
+                    setattr(
+                        archive.results.properties.catalytic.catalyst,
+                        catalyst_attr,
+                        [value],
+                    )
+
+    def add_referencing_methods(
+            self, archive: 'EntryArchive', logger: 'BoundLogger', number=10) -> None:
+        '''
+        This function looks for other entries that reference the sample and checks the
+        entry type and if it finds a ELNXRayDiffration entry it adds 'XRD' to the
+        characterization_methods in result.
+
+        Args:
+            archive (EntryArchive): The archive containing the section that is being
+            normalized.
+            logger('Bound Logger'): A structlog logger.
+            number: specifies the number of referencing entries that are checked,
+            set to 10 by default
+        '''
+        catalyst_sample = self.m_root().metadata.entry_id
+
+        if self.lab_id is None:
+            logger.warn(f'Found no entries with reference: "{catalyst_sample}".')
+            return
+
+        from nomad.search import MetadataPagination, search
+
+        query = {'entry_references.target_entry_id': catalyst_sample}
+        search_result = search(
+            owner='all',
+            query=query,
+            pagination=MetadataPagination(page_size=number),
+            user_id=archive.metadata.main_author.user_id,
+        )
+
+        if search_result.pagination.total > 0:
+            methods = []
+            for entry in search_result.data:
+                if entry['entry_type'] == 'ELNXRayDiffraction':
+                    method = 'XRD'
+                    methods.append(method)
+                elif entry['entry_type'] == 'CatalystCollection':
+                    pass
+                elif entry['entry_type'] == 'CatalystSampleCollection':
+                    pass
```

Comment:
This can now be modified since you always get an Activity with a method quantity (could be None, in that case use m_def.name)

Reply:
I changed to 'entry_type'  if the results.eln.methods is not specified further beyond 'ELNMeasurement'